### ISSUE TITLE
Rework design for integrated charger/5V boost chip

### ISSUE THREAD

**Issue by amcewen** [mcqn/my-bikes-got-led#20]
TI have [a nice integrated battery charging and 5V boost chip](https://www.ti.com/lit/ds/symlink/bq25895m.pdf) that could replace both the charger and the 5V boost circuit.

Nice things about it are:
* 3.1A current from the 5V boost circuit, which would let us run more lights from it (currently it's ~1.5A IIRC, though that might include a bit of headroom so we're not running it at max output all the time)
* Fewer chips needed, so might be smaller
* Ability to charge at higher currents depending on USB charger attached

It'd be useful to compare the costs - I think it'll be more expensive than the current option, but it'd be useful to know roughly how much (so what's the total BoM cost for this circuit over the existing charger and 5V boost, inc. inductors, etc.); and maybe do a bit of a search through LCSC to see if there are any alternative all-in-one options.  Presumably it's a fairly popular circuit, given how many USB power banks there are.

**Comment by huffeec** [comment]
Very ball park numbers gives a ~+$0.24 difference with this chip over the current setup.
I've not managed to find anything equivalent on LCSC. (finding anything on LCSC is a challenge though so I'll have another quick look at some point)  

Old  
Battery management – CN3791 $0.5
        + Power mosfet C308748 $0.25
Battery protection – FS312F-G $0.12 (discontinued at LCSC)
	+ Mosfets FS8205 $0.18 (discontinued at LCSC)	
Boost converter – FP6276B $0.34
	+ inductor – C27442 $0.16

Total - **$1.55**

New 
BQ25895M **$1.79**

**Comment by amcewen** [comment]
Dismantling one of the Pimoroni battery packs, they've got a protection PCB internally, so dropping the battery protection should be okay.

There's a 2.2uH inductor on the sample circuit for the new chip, but that shouldn't make a massive difference - there seem to be some on LCSC that are super cheap ($0.03!) and even if I'm missing some of the more important criteria (was just a quick search for shielded ones, at 2.2uH with a 3.5A+ current rating) they'll likely not be massively more expensive.

Might be worth a quick scan of Kicad to see if that throws up any alternative options; maybe Tindie too?  But if we don't find anything cheaper I think a move to the BQ is worth doing.

**Comment by huffeec** [comment]
[d6158d3](https://github.com/mcqn/my-bikes-got-led/commit/d6158d3ef8e8aa7f31227338dbe93214acebe4ec) has a rework that includes the battery management from #25 

Sanity checking before layout needed:

* Is the switch circuit ok? It takes the JSM2302 Mosfet from the earlier versions, but without the battery protection chip. - Batt denotes all battery management side grounding, Gnd everything on the other side of the switch.
* Do we need the external 32.68KHz crystal for the c6 mini? The [datasheet for the module](https://www.espressif.com/sites/default/files/documentation/esp32-c6-mini-1_mini-1u_datasheet_en.pdf) pg32 and the schematic for the [dev kit ](https://docs.espressif.com/projects/esp-dev-kits/en/latest/_static/esp32-c6-devkitm-1/schematics/esp32-c6-devkitm-1-schematics.pdf)have it. I can't find much of a discussion about it though.

**Comment by amcewen** [comment]
The [C6 datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-c6_datasheet_en.pdf) lists that it includes a 40MHz crystal oscillator.  It also talks about the 32KHz crystal and points to [the Reset and Clock chapter in the ESP32-C6 Technical Reference Manual](https://www.espressif.com/sites/default/files/documentation/esp32-c6_technical_reference_manual_en.pdf#section.8) (TBF the datasheet seems to be almost a 50-odd page index for the technical reference manual :rofl:)

Section 8.2.4.2 seems the most relevant:

> The clock sources for LP_SLOW_CLK and LP_FAST_CLK are low-frequency clocks:
> * LP_SLOW_CLK can be derived from:
>   * RC_SLOW_CLK
>   * XTAL32K_CLK
>   * RC32K_CLK
>   * OSC_SLOW_CLK

Reading through that, it seems that the external 34KHz crystal is optional, because it's possible to use the internal 32KHz RC oscillator instead.  Presumably it's available to allow a more precise source rather than an RC circuit.  I'm thinking we skip it, but maybe include a couple of test points near the pins so we could tack one in (on the prototype at least?) if we have any problems

**Comment by amcewen** [comment]
The switch circuit looks okay to me.  The battery packs we use have a protection circuit built-in, so I think it's correct to omit that from this version of the board.

**Comment by huffeec** [comment]
[945ffdf](https://github.com/mcqn/my-bikes-got-led/commit/945ffdfe1df8e9bcc706f4c1dc4ae111a14a96aa) check in new design with charger/5V boost

**Comment by amcewen** [comment]
Some review notes.

Charger circuit looks good.

Probably not a showstopper but do we still need two polyfuses?  I'm thinking it might be better to just have one between the boost 5V output and everything else - so it'd protect the 3V3 regulator (and so ESP), neopixels, plus any shorting of the breakout pins.

We've got a chunky trace for the 5V until it gets to `F2` and then the route from there to the LED connectors - `J2` and `J3` is a thin trace.  Maybe if we route all of the `5V` through `F1` and lose `F2`, then we could shift the `5V`/`GND`/`3V3` breakout pins nearer to the connector side, and route the `LED_Data` and `LED_Clk` lines on the back copper to the left of the breakouts rather than the right?  Then there'd be room for a fat `5V` trace on the back copper from the breakout to `J3`?

Haven't reviewed the ESP32-C6 circuit yet, but have covered everything else.

**Comment by huffeec** [comment]
I have removed F2, DRC still passes.

On My Baby's the Enable pin is pulled high, the switch pulls reset and IO4 low

On the C6 there is no reset pin, the switch pulls enable low, via RC.
Boot mode is selected via gpio 08/09
In the [default circuit (page 32)](https://www.espressif.com/sites/default/files/documentation/esp32-c6-mini-1_mini-1u_datasheet_en.pdf) 08 is pulled high 09 is pulled up by internal resistor or pulled low via a jumper.

Table 3.3.2 Chip Boot Mode Control of the datasheet linked above shows this gives default SPI boot mode or Download boot if the jumper is connected.

Are we ok with this default behaviour or do we need to tie a gpio pin to the switch for the mod?

**Comment by amcewen** [comment]
How would a non-techie user clear all their info from the board, if they sold/gave it to someone else?  How would anyone wipe the WiFi details if it was set to create an access point but you'd forgotten the password for it?

I don't mean for that to sound confrontational, I'm just laying out the reasons - I think they're the key ones - that I added the factory reset feature originally.  If there are other ways that we can achieve them then we might not need it any more.

Otherwise, if we'd need to do a bit of experimenting because there's now an RC circuit in the mix on the Enable pin, could we add a test point on that pin to let us breadboard the extra parts we'd need for a factory reset on the next rev?

**Comment by huffeec** [comment]
Agreed, I guess I am asking two different questions at once.

Do we need to be able to toggle between SPI/Download boot mode? If so do we need a solderable jumper or pin headers?

How to implement the factory reset behaviour on this rev. My Baby's pulls reset high when the switch is pressed (Enable always on) C6 uses the switch to pull Enable low. Is a change of state enough on the factory reset GPIO? 
For this rev is it enough to have a gpio connected to the switch via a breakable jumper and a test point on Enable? ie no additional RC for the GPIO?

**Comment by amcewen** [comment]
Good point about boot mode.  If there's room for some sort of pin access for bootmode I think that'd be useful - with MiaB we've seen that sometimes the program-over-USB doesn't work properly and putting it into boot mode explicitly gets it out of that state.

And on the factory reset side, I don't think it matters which way the GPIO is pulled (though we'd need to update the usermod in WLED if it changes), but the board needs to still boot up if the switch is held closed rather than stay in reset.  I can't quite work out what the circuit would look like without breadboarding it to double-check, so I think a test  point on Enable would let us mod one of the prototype PCBs to work that out.  I'm not sure connecting the GPIO with a breakable jumper would be enough - feels like we might need to re-jig and/or add to the caps and resistor around there - so we can test with one of the broken-out-for-general-use GPIOs (and then pick a not-broken-out one for the proper implementation).

I guess if we're (potentially) desoldering one (or more) of `C12`, `C13` or `R11` for testing that, maybe soldering an extra wire to one of them would be as easy as a test point?  It'll only be on one board after all.  Feels like I'm talking myself out of any changes to this rev, and maybe it's just a new issue to address once we've got the hardware? :grinning:

**Comment by huffeec** [comment]
If it helps make this decision. I've uploaded the files to JLC with a gpio connected to the switch and some test points in that region for hacking if needed. Just to gauge costs if nothing else...

For five boards delivered...... $150   Includes $25 setup fee, $50 loading fee $10/ea components.

I may have misremembered but it feels like they are getting less competitive...

It's left in the basket to think about

I've also learned that LCSC stock is not equal to JLC stock anymore. 
So this has a [Molex](https://www.lcsc.com/product-detail/USB-Connectors_MOLEX-1054500101_C134092.html) USB connector instead of the [Amphenol](https://www.lcsc.com/product-detail/USB-Connectors_Amphenol-ICC-12401610E4-2A_C5119948.html)

**Comment by amcewen** [comment]
We've got a two-pin header for `boot` mode; and have been happy with the design so far (plus WLED has an option for resetting the WiFi settings with the default switch GPIO, so I don't think we need our factory reset usermod any more).
